lib/enum/enum.rb

```diff
@@ -14,6 +14,20 @@ def enum_accessor(*field_names)
         reload
       end
 
+      def enum(columns)
+        columns.each do |column, values|
+          values = values.zip(values.map(&:to_s)).to_h if values.is_a?(Array)
+          values.each do |method, value|
+            class_eval <<~METHOD, __FILE__, __LINE__ + 1
+              # frozen_string_literal: true
+              def #{method}?
+                #{column} == #{value.inspect}
+              end
+            METHOD
```

Comment:
Great call on `fronzen_string_literal` and the line number.

I had taken my inspiration from https://github.com/rails/rails/blob/main/activesupport/lib/active_support/core_ext/class/attribute.rb#L119-L120

They only call `class_eval` once, hence the `method_definitions`, and they use the caller location. But this should work fine.

Interestingly, enum uses `def_method`.

^ more of a note to future self.